src/Tickets/Commerce/Order_Modifiers/Custom_Tables/Order_Modifier_Relationships.php

```diff
@@ -55,66 +61,23 @@ class Order_Modifier_Relationships extends Table {
 	/**
 	 * An array of all the columns in the table.
 	 *
-	 * @since 5.20.0
+	 * @since TBD
 	 *
 	 * @var string[]
 	 */
-	public static function get_columns(): array {
-		return [
-			'id',
-			'modifier_id',
-			'post_id',
-			'post_type',
-		];
-	}
+	public static function get_schema_history(): array {
+		$table_name = self::table_name( true );
 
-	/**
-	 * Returns the table creation SQL in the format supported
-	 * by the `dbDelta` function.
-	 *
-	 * @since 5.18.0
-	 *
-	 * @return string The table creation SQL, in the format supported
-	 *                by the `dbDelta` function.
-	 */
-	protected function get_definition() {
-		global $wpdb;
-		$table_name      = self::table_name( true );
-		$charset_collate = $wpdb->get_charset_collate();
-
-		return "
-			CREATE TABLE `$table_name` (
-				`id` BIGINT UNSIGNED NOT NULL AUTO_INCREMENT,
-				`modifier_id` BIGINT UNSIGNED NOT NULL,
-				`post_id` BIGINT UNSIGNED NOT NULL,
-				`post_type` VARCHAR(20) NOT NULL,
-				PRIMARY KEY (`id`)
-			) $charset_collate;
-		";
-	}
-
-	/**
-	 * Allows extending classes that require it to run some methods
-	 * immediately after the table creation or update.
-	 *
-	 * @since 5.18.0
-	 *
-	 * @param array<string,string> $results A map of results in the format
-	 *                                      returned by the `dbDelta` function.
-	 *
-	 * @return array<string,string> A map of results in the format returned by
-	 *                              the `dbDelta` function.
-	 */
-	protected function after_update( array $results ): array {
-		// If nothing was changed by dbDelta(), bail.
-		if ( ! count( $results ) ) {
-			return $results;
-		}
-
-		// Helper method to check and add indexes.
-		$results = $this->check_and_add_index( $results, 'tec_order_modifier_relationship_index_modifier_id', 'modifier_id' );
-		$results = $this->check_and_add_index( $results, 'tec_order_modifier_relationship_index_post_id', 'post_id' );
+		return [
+			self::SCHEMA_VERSION => function () use ( $table_name ) {
+				$columns   = new Column_Collection();
```

Comment:
This is a little out of scope for this PR, but after looking at a few of these I can't help but think that it would be nice for `Column_Collection` (or event the base `Collection` class) to have a static method that accepts the collection items and returns the new collection. E.g. something like this:

```php
$columns = Column_Collection::create(
	[
		new ID( 'id' ),
		new Referenced_ID( 'modifier_id' ),
		new Referenced_ID( 'post_id' ),
		( new String_Column( 'post_type' ) )->set_length( 20 ),
	]
);
```

Not strictly needed, but IMO it would be a bit nicer. That said, I can understand that this but be too much out of scope for now.